### PR TITLE
Add support for more elliptic curves

### PR DESCRIPTION
**Pull request by sosthene-nitrokey**
- [x] p348 (using RustCrypto)
- [x] p521 (using Ruscrypto)
- [x] brainpoolP256r1 (Mechanism definition only, implementation in separate backend)
- [x] brainpoolP384r1 (Mechanism definition only, implementation in separate backend)
- [x] brainpoolP512r1  (Mechanism definition only, implementation in separate backend)

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -110,6 +113,8 @@ hmac-sha1 = []
 hmac-sha256 = []
 hmac-sha512 = []
 p256 = []
+p384 = ["dep:p384"]
+p521 = ["dep:p521", "ecdsa"]
```
Comment (robin-nitrokey) [inline Cargo.toml:117]:
```suggestion
p521 = ["dep:p521", "dep:ecdsa"]
```

**Comment by sosthene-nitrokey** [comment]
This will need to get rebased on top of trussed-main to remove the nitrokey main commits.

File: Cargo.toml [hunk 2]
```diff
@@ -57,6 +57,9 @@ littlefs2 = "0.4.0"
 p256-cortex-m4 = { version = "0.1.0-alpha.6", features = ["prehash", "sec1-signatures"] }
 salty = { version = "0.3.0", features = ["cose"] }
 serde-indexed = "0.1.0"
+p384 = { version = "0.13.0", optional = true, default-features = false, features = ["sha384", "ecdh", "ecdsa"] }
+p521 = { version = "0.13.3", optional = true, default-features = false, features = ["sha512", "ecdh", "ecdsa"] }
+ecdsa = { version = "0.16.9", optional = true, default-features = false }
```
Comment (robin-nitrokey) [inline Cargo.toml:62]: nit: move to the RustCrypto section

**Review by robin-nitrokey — COMMENTED** [review]
Can you please rebase the branch?  It’s hard to review together with the other changes from nitrokey/main.

**Comment by sosthene-nitrokey** [comment]
This branch is actually the rebased version. This PR should be against the nitrokey fork

**Pull request closed without merging**